File: app/src/main/java/com/chatter/furrki/chatter/RoomListViewAdapter.java
package com.chatter.furrki.chatter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.parse.ParseUser;

import java.util.ArrayList;

public class RoomListViewAdapter extends ArrayAdapter<ParseUser> {

    private final LayoutInflater inflater;
    private final Context context;
    private ViewHolder holder;
    private final ArrayList<ParseUser> persons;

    public RoomListViewAdapter(Context context, ArrayList<ParseUser> persons) {
        super(context,0, persons);
        this.context = context;
        this.persons = persons;
        inflater = LayoutInflater.from(context);
    }

    @Override
    public int getCount() {
        return persons.size();
    }

    @Override
    public ParseUser getItem(int position) {
        return persons.get(position);
    }

    @Override
    public long getItemId(int position) {
        return persons.get(position).hashCode();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        if (convertView == null) {

            convertView = inflater.inflate(R.layout.room_list_item, null);

            holder = new ViewHolder();
            holder.userimage = (ImageView) convertView.findViewById(R.id.user_image);
            holder.username = (TextView) convertView.findViewById(R.id.user_name);
            convertView.setTag(holder);

        }
        else{
            //Get viewholder we already created
            holder = (ViewHolder)convertView.getTag();
        }

        ParseUser person = persons.get(position);
        if(person != null){
            //holder.userimage.setImageResource();
            holder.username.setText(person.getUsername());

        }
        return convertView;
    }

    //View Holder Pattern for better performance
    private static class ViewHolder {
        TextView username;
        ImageView userimage;

    }
}

File: app/src/main/java/com/chatter/furrki/chatter/RoomListViewAdapter.kt
package com.chatter.furrki.chatter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView

import com.parse.ParseUser

import java.util.ArrayList

class RoomListViewAdapter(context: Context, private val rooms: ArrayList<Room>) : ArrayAdapter<Room>(context, 0, rooms) {

    private val inflater: LayoutInflater
    private var holder: ViewHolder? = null

    init {
        inflater = LayoutInflater.from(context)
    }

    override fun getCount(): Int {
        return rooms.size
    }

    override fun getItem(position: Int): Room? {
        return rooms[position]
    }

    override fun getItemId(position: Int): Long {
        return rooms[position].hashCode().toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView

        if (convertView == null) {

            convertView = inflater.inflate(R.layout.room_list_item, null)

            holder = ViewHolder()
            holder!!.userimage = convertView!!.findViewById<View>(R.id.user_image) as ImageView
            holder!!.username = convertView.findViewById<View>(R.id.user_name) as TextView
            convertView.tag = holder

        } else {
            //Get viewholder we already created
            holder = convertView.tag as ViewHolder
        }

        val room = rooms[position]
        if (room != null) {
            //holder.userimage.setImageResource();
            val members = room.getMembers()
            var other = ParseUser.getCurrentUser()

            if (members[0].objectId == other.objectId) {
                other = members[1]
            } else {
                other = members[0]
            }
            other.fetchIfNeeded()


            holder!!.username!!.setText(other.getUsername())

        }
        return convertView
    }

    //View Holder Pattern for better performance
    private class ViewHolder {
        internal var username: TextView? = null
        internal var userimage: ImageView? = null

    }
}